pkg/cmd/gist/create/create.go

```diff
@@ -91,13 +92,23 @@ func createRun(opts *CreateOptions) error {
 		fileArgs = []string{"-"}
 	}
 
-	files, err := processFiles(opts.IO.In, fileArgs)
+	files, nonGistFilesSorted, err := processFiles(opts.IO.In, fileArgs)
```

Comment:
My suggestion would be to avoid overloading `processFiles` with this functionality and keep it like before.

Instead, you could add a new function called `guessGistName` that takes in the `files` map and returns the name for this gist, or a blank string if there are no files other than `gistfile*`.

Then you can test that function in isolation and you don't have to add additional test cases nor fixture files to `Test_processFiles`.

Reply:
I removed the extra fixture file and test case to `Test_createRun` that I added.

I implemented the `guessGistName` function you suggested. I think that is a much better approach.